meta-cube/recipes-support/overc-system-agent/files/overc-system-agent-1.2/Overc/overc.py

```diff
@@ -230,7 +230,12 @@ def _container_delete(self, container, template, force):
         self.message = self.container.message
 
     def container_upgrade(self):
-        self._container_upgrade(self.args.name, self.args.template, self.args.rpm, self.args.image)
+        # Perform overlay check fist
+        if self.container.pre_upgrade(self.args.name, self.args.template) is 1:
+            print "Container %s got overlayed dir, must use system upgrade." % self.args.name
```

Comment:
This should be: "Container references overlay directories, a system upgrade must be performed". Also, we should dump the overlay directories here. The more information we make available, the more understandable the message will be.